lib/elixir/lib/module.ex

```diff
@@ -1158,6 +1273,18 @@ defmodule Module do
         :ok
     end
 
+    case {key, value} do
+      {:impl, nil} -> :ok
+      {:impl, false} -> :ok
+      {:impl, true} -> raise "TODO: implement"
```

Comment:
@samphilipd I am already decided on making `@impl true` part of the implementation so we should go ahead with it.